front-end/src/domains/game-board/components/game-board/game-board.component.tsx

```diff
@@ -0,0 +1,195 @@
+import {GameHex} from "./game-hex.component";
+import React from "react";
+
+type Node = {
+  type: NodeType;
+}
+
+type NodeType = "water" | "space" | "concrete" | "fuel" | "food" | "player" | "metal" | "empty";
```

Comment:
Look at this guy using TypeScript like a champ!

Another thing you can look into is a TS Enum, which is nice when you want to harden these string values and export them different places.

```suggestion
enum NodeType {
   WATER,
   SPACE,
   CONRETE,
   FUEL,
}
```

Reply:
We could even start a `constants` file for all of these hardcoded strings 😉.